CppCoreGuidelines.md

```diff
@@ -7322,7 +7322,51 @@ But heed the warning: [Avoid "naked" `union`s](#Ru-naked)
 
 ##### Example
 
-    ??? short-string optimization; safe union without dscriminant ???
+    // Short string optimization
+    
+    constexpr size_t buffer_size = 16;
+    
+    class Immutable_string
+    {
+    public:
+        Immutable_string(const char* str) :
+            size(strlen(str))
+        {
+            if (size < buffer_size)
+            {
+                strcpy(string_buffer, str);
+            }
+            else
+            {
+                string_ptr = new char[size + 1];
+                strcpy(string_ptr, str);
+            }
+        }
+    
+        ~Immutable_string()
+        {
+            if (size >= buffer_size)
+            {
+                delete string_ptr;
+            }
+        }
```

Comment:
Style nit: Branches of if-statements that are single statements go by themselves, no need to package them in braces.